src/modules/navigation/NavigationState.js

```diff
@@ -14,70 +17,83 @@ export function switchTab(index) {
 }
 
 // Action creators
-export function pushRoute(state) {
-  return (dispatch, getState) => {
-    // conditionally execute push to avoid double
-    // navigations due to impatient users
-    if (!isNavigationAnimationInProgress(getState())) {
-      dispatch({type: PUSH_ROUTE, payload: state});
-    }
+export function pushRoute(route) {
+  return {
+    type: PUSH_ROUTE,
+    payload: route
   };
 }
 
 export function popRoute() {
   return {type: POP_ROUTE};
 }
 
-export function navigationCompleted() {
-  return {type: NAVIGATION_COMPLETED};
-}
-
-const initialState = fromJS(
-  createNavigationState('MainNavigation', 'App', [
-    createNavigationState('HomeTab', 'Home', [{key: 'Counter', title: 'Counter'}]),
-    createNavigationState('ProfileTab', 'Profile', [{key: 'Color', title: 'Color'}])
-  ]));
+// reducers for tabs and scenes are separate
+const initialState = fromJS({
+  tabs: {
+    index: 0,
+    routes: [
+      {key: 'HomeTab', title: 'HOME'},
+      {key: 'ProfileTab', title: 'PROFILE'}
+    ]
+  },
+  // Scenes for the `HomeTab` tab.
+  HomeTab: {
+    index: 0,
+    routes: [{key: 'Counter', title: 'Counter Screen'}]
+  },
+  // Scenes for the `ProfileTab` tab.
+  ProfileTab: {
+    index: 0,
+    routes: [{key: 'Color', title: 'Color Screen'}]
+  }
+});
 
```

Comment:
Learning from other example from the community and `react-native` NavigationExperimental examples, it results much clearer to maintain two separate states for `tabs` and `scenes`. Also `key` on the state is not used anymore as of `0.28`.